### PR TITLE
Leonardo Romero | integration-api

### PR DESCRIPTION
**Pull request by leoromero97**
## Summary
- Create service post
- Add logic and integratino with custom hook in Login screen
- Add QueryClientProvider for wrapper myApp and create queryClient from react query
- Create IUser interface
- Create useLogin() custom hook
- Modify regex to accept password from api
- Create api config to baseURL
- Add axios and react query libraries

## Screenshots
N/A

### CODE REVIEW

File: config/api.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios';
+
+const api = axios.create({
+  baseURL: 'https://reqres.in/api'
```
Comment (jjuli93) [inline config/api.ts]: In a real app we would have many environments (development, staging, production) so the base api url should change for each environment. That's why baseURL should be and environment variable.

File: pages/Login/index.tsx [hunk 2]
```diff
@@ -19,8 +20,14 @@ const Login: NextPage = () => {
     formState: { errors },
   } = useForm<ILogin>();
   const router = useRouter();
+  const { data, mutate, ...dataQuery } = useLogin();
 
-  const onSubmit: SubmitHandler<ILogin> = (data) => router.push("/Home");
+  const onSubmit: SubmitHandler<ILogin> = (dataForm) => {
+    const success = (token: any) => {
+      router.push("/Home", token);
```
Comment (jjuli93) [inline pages/Login/index.tsx]: all route names should be centralized in a file to make it easier to change a route name

File: pages/Login/index.tsx [hunk 3]
```diff
@@ -19,8 +20,14 @@ const Login: NextPage = () => {
     formState: { errors },
   } = useForm<ILogin>();
   const router = useRouter();
+  const { data, mutate, ...dataQuery } = useLogin();
 
-  const onSubmit: SubmitHandler<ILogin> = (data) => router.push("/Home");
+  const onSubmit: SubmitHandler<ILogin> = (dataForm) => {
+    const success = (token: any) => {
```
Comment (jjuli93) [inline pages/Login/index.tsx]: try not to use any. In this case we can use IUserLoginResponse

**Review by jjuli93 — CHANGES_REQUESTED** [review]

File: components/ProgressBar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+export interface IProgressBar {
+  value: number;
+  max: number;
+  message: string;
+}
+
+function ProgressBar({ value, max, message }: IProgressBar) {
+  return (
+    <div>
```
Comment (jjuli93) [inline components/ProgressBar/index.tsx]: is this div needed?

File: pages/Login/index.tsx [hunk 5]
```diff
@@ -22,106 +25,138 @@ const Login: NextPage = () => {
...
   const { data, mutate, ...dataQuery } = useLogin();
+  const [isError, setError] = useState(false);
+  const [value, setValue] = useState(0);
 
   const onSubmit: SubmitHandler<ILogin> = (dataForm) => {
     const success = () => {
       router.push(Routes.Home);
     };
-    mutate(dataForm, { onSuccess: success });
+    const error = () => {
```
Comment (jjuli93) [inline pages/Login/index.tsx]: no need for this, react query already provides a boolean for this in the useMutation return object

File: pages/Login/index.tsx [hunk 6]
```diff
@@ -22,106 +25,138 @@ const Login: NextPage = () => {
   } = useForm<ILogin>();
   const router = useRouter();
   const { data, mutate, ...dataQuery } = useLogin();
+  const [isError, setError] = useState(false);
+  const [value, setValue] = useState(0);
```
Comment (jjuli93) [inline pages/Login/index.tsx]: use a more explicit name. what does the value represents?

File: pages/Login/index.tsx [hunk 7]
```diff
@@ -22,106 +25,138 @@ const Login: NextPage = () => {
...
       router.push(Routes.Home);
     };
-    mutate(dataForm, { onSuccess: success });
+    const error = () => {
+      setError(true);
+    };
+    mutate(dataForm, { onSuccess: success, onError: error });
   };
 
+  useEffect(() => {
```
Comment (jjuli93) [inline pages/Login/index.tsx]: if we take this approach, we will have to copy this code every time we want to show the loader, right?

File: pages/ErrorScreen/index.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import emMonocle from '../../public/assets/emojis/monocle.svg';
+import styles from './styles.module.css';
+
+export interface IErrorScreen {
+  className?: string;
+  message: string;
+  title: string;
+}
+
+function ErrorScreen({className, message, title}: IErrorScreen) {
```
Comment (jjuli93) [inline pages/ErrorScreen/index.tsx]: Is this a component or a page?
it is on the pages folder but it is being used as a component

**Review by jjuli93 — CHANGES_REQUESTED** [review]

File: components/ProgressBar/index.tsx (hunk 4 above)
Comment (leoromero97) [reply components/ProgressBar/index.tsx]: No need

File: pages/Login/index.tsx (hunk 5 above)
Comment (leoromero97) [reply pages/Login/index.tsx]: Okkk

File: pages/ErrorScreen/index.tsx (hunk 8 above)
Comment (leoromero97) [reply pages/ErrorScreen/index.tsx]: is componenttt

**Review by jjuli93 — APPROVED** [review]

**Pull request merged**